docs/DeveloperGuide.md

```diff
@@ -273,6 +273,83 @@ the gender is not of valid format; Invalid person exception is thrown if the per
 contact list. Error message is displayed on the GUI subsequently.
 </div>
 
+### Add Date Of Birth
+
+The Add Date Of Birth feature allows users to add a date of birth field (format: dd/mm/yyyy) to a person in the contact list. It is performed as a part of `addPersonCommand#execute()`.
+
+These operations are exposed in the `Model` interface as the method `Model#addPerson()`, which calls
+`AddressBook#addPerson()` which calls `UniquePersonList#add()` to add a new person in the person list
+stored in AddressBook.
+
+The following sequence diagram shows the methods calls related to add person operation:
+
+![AddPersonSequenceDiagram](images/AddPersonSequenceDiagram.png)
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** `cmd` in the diagram represents the add
+person command text entered by user. The specific `UniquePersonList` operations are not shown in the diagram
+for simplicity.
+</div>
+
+The following activity diagram shows what happens when a user executes a new add command:
+
+![AddPersonActivityDiagram](images/AddPersonActivityDiagram_DOB.png)
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** Only the
+activities related to date of birth field are considered and shown in this activity diagram.
+</div>
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:**Parser exceptions are thrown and caught if
+date of birth field is not provided in the command, or the date of birth is not of valid format; Duplicated person exception is
+thrown if the person to add already exists in the contact list. Error message is displayed on the GUI subsequently.
+</div>
+
+#### Design considerations:
+
+**Aspect: Whether date of birth field should be optional for a person:**
+
+* **Alternative 1 (current choice):** Compulsory date of birth field:
+    * Pros: It is a more logical implementation because date of birth is a common attribute for all persons,
+  similar to name, address, etc, which are also compulsory field for persons in contact list.
```

Comment:
Minor edit: should be plural
```suggestion
    * Pros: It is a more logical implementation because date of birth is a common attribute for all persons,
  similar to name, address, etc, which are also compulsory fields for persons in contact list.
```